### ISSUE TITLE
New batch_domains method

### ISSUE THREAD

**Issue by mattias-p** [zonemaster/zonemaster-backend#1151]
Depends on #960.

We should add a new method to the RPCAPI.

Name: batch_domains

Input:
* `batch_id` - A Batch ID. The batch to examine.
* `states` - An optional list of elements from {"waiting", "running", "completed", "cancelled", "crashed"}. Default: ["waiting", "running", "completed", "cancelled", "crashed"]. The set of properties to include in the output.

Output:
* `waiting` - An optional object mapping domain names to job ids. The set of waiting jobs in the batch.
* `running` - An optional object mapping domain names to job ids. The set of running jobs in the batch.
* `completed` - An optional object mapping domain names to job ids. The set of completed jobs in the batch.
* `cancelled` - An optional object mapping domain names to job ids. The set of cancelled jobs in the batch.
* `crashed` - An optional object mapping domain names to job ids. The set of crashed jobs in the batch.

N.B. This is basically just an updated version of the get_batch_list method from #1119 that's been renamed and adapted to zonemaster/zonemaster-gui#460. I feel there are too many ideas in #1119 and it's better to split this part out to its own issue.
